From c97e83a7e8f8913462fd4ad0768b423ae9ecb809 Mon Sep 17 00:00:00 2001
From: Sun Haiyong <sunhaiyong@zdbr.net>
Date: Mon, 5 May 2025 08:07:08 +0000
Subject: [PATCH] q4wine 1.4.2 fix for l10n files install failed.

---
 src/l10n/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/l10n/CMakeLists.txt b/src/l10n/CMakeLists.txt
index 7085a6f..5bdc97d 100644
--- a/src/l10n/CMakeLists.txt
+++ b/src/l10n/CMakeLists.txt
@@ -24,7 +24,7 @@ get_directory_property(Q4WINE_LIB_SOURCES DIRECTORY ../q4wine-lib DEFINITION Q4W
 
 set_source_files_properties(
     ${TS_FILES}
-    PROPERTIES OUTPUT_LOCATION "${CMAKE_SOURCE_DIR}/src/l10n"
+    PROPERTIES OUTPUT_LOCATION "${CMAKE_CURRENT_BINARY_DIR}"
 )
 
 qt_add_translations(q4wine
-- 
2.31.1

