From d474d8e0e3963623991941a4cba9c811aa8d7e53 Mon Sep 17 00:00:00 2001
From: Sun Haiyong <sunhaiyong@zdbr.net>
Date: Sat, 25 Nov 2023 13:02:58 +0000
Subject: [PATCH] c-utf8 set version-scripts to yes for default.

---
 src/meson.build | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/meson.build b/src/meson.build
index baeecb7..d6a9033 100644
--- a/src/meson.build
+++ b/src/meson.build
@@ -19,9 +19,9 @@ libcutf8_both = both_libraries(
         ],
         dependencies: libcutf8_deps,
         install: not meson.is_subproject(),
-        link_args: dep_cstdaux.get_variable('version-scripts') == 'yes' ? [
+        link_args: [
                 '-Wl,--version-script=@0@'.format(libcutf8_symfile),
-        ] : [],
+        ],
         link_depends: libcutf8_symfile,
         soversion: 0,
 )
-- 
2.31.1

