--- rustc-1.66.0-src.orig/compiler/rustc_llvm/llvm-wrapper/PassWrapper.cpp	2023-01-09 08:44:39.785578960 +0000
+++ rustc-1.66.0-src/compiler/rustc_llvm/llvm-wrapper/PassWrapper.cpp	2023-01-09 08:21:34.605065300 +0000
@@ -225,7 +225,7 @@ static Optional<CodeModel::Model> fromRu
   case LLVMRustCodeModel::Large:
     return CodeModel::Large;
   case LLVMRustCodeModel::None:
-    return None;
+    return std::nullopt;
   default:
     report_fatal_error("Bad CodeModel.");
   }
--- rustc-1.66.0-src.orig/compiler/rustc_llvm/llvm-wrapper/RustWrapper.cpp	2023-01-09 08:44:39.719579048 +0000
+++ rustc-1.66.0-src/compiler/rustc_llvm/llvm-wrapper/RustWrapper.cpp	2023-01-09 08:26:30.721033443 +0000
@@ -317,7 +317,7 @@ extern "C" LLVMAttributeRef LLVMRustCrea
 }
 
 extern "C" LLVMAttributeRef LLVMRustCreateAllocSizeAttr(LLVMContextRef C, uint32_t ElementSizeArg) {
-  return wrap(Attribute::getWithAllocSizeArgs(*unwrap(C), ElementSizeArg, None));
+  return wrap(Attribute::getWithAllocSizeArgs(*unwrap(C), ElementSizeArg, std::nullopt));
 }
 
 #if LLVM_VERSION_GE(15, 0)
@@ -671,7 +671,7 @@ enum class LLVMRustChecksumKind {
 static Optional<DIFile::ChecksumKind> fromRust(LLVMRustChecksumKind Kind) {
   switch (Kind) {
   case LLVMRustChecksumKind::None:
-    return None;
+    return std::nullopt;
   case LLVMRustChecksumKind::MD5:
     return DIFile::ChecksumKind::CSK_MD5;
   case LLVMRustChecksumKind::SHA1:
