From 91f2e10f1d9988b896c134ba2fac398b5dede209 Mon Sep 17 00:00:00 2001
From: Sun Haiyong <sunhaiyong@zdbr.net>
Date: Sun, 19 Nov 2023 03:33:49 +0000
Subject: [PATCH] kdoctools fix build error for libxml2 2.12.0.

---
 src/xslt.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/xslt.cpp b/src/xslt.cpp
index 28c724f..dbc5435 100644
--- a/src/xslt.cpp
+++ b/src/xslt.cpp
@@ -14,6 +14,7 @@
 #include <libxml/parser.h>
 #include <libxml/parserInternals.h>
 #include <libxml/xmlIO.h>
+#include <libxml/xmlsave.h>
 #include <libxslt/transform.h>
 #include <libxslt/xsltInternals.h>
 #include <libxslt/xsltconfig.h>
-- 
2.31.1

