From b5e1f931ff2b820c8201f37a519992c046cce51d Mon Sep 17 00:00:00 2001
From: Sun Haiyong <sunhaiyong@zdbr.net>
Date: Fri, 16 Aug 2024 01:19:19 +0000
Subject: [PATCH] wine fix 39bit patch 2.

---
 dlls/ntdll/unix/virtual.c | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/dlls/ntdll/unix/virtual.c b/dlls/ntdll/unix/virtual.c
index 480eef9..73ac445 100644
--- a/dlls/ntdll/unix/virtual.c
+++ b/dlls/ntdll/unix/virtual.c
@@ -2479,7 +2479,6 @@ static NTSTATUS map_pe_header( void *ptr, size_t size, int fd, BOOL *removable )
     return STATUS_SUCCESS;  /* page protections will be updated later */
 }
 
-#ifdef __aarch64__
 
 /***********************************************************************
  *           get_host_addr_space_limit
@@ -2507,6 +2506,7 @@ static void *get_host_addr_space_limit(void)
     return (void *)((addr << 1) - (granularity_mask + 1));
 }
 
+#ifdef __aarch64__
 
 /***********************************************************************
  *           alloc_arm64ec_map
@@ -3342,12 +3342,12 @@ void virtual_init(void)
     pthread_mutex_init( &virtual_mutex, &attr );
     pthread_mutexattr_destroy( &attr );
 
-#ifdef __aarch64__
+// #ifdef __aarch64__
     host_addr_space_limit = get_host_addr_space_limit();
     TRACE( "host addr space limit: %p\n", host_addr_space_limit );
-#else
-    host_addr_space_limit = address_space_limit;
-#endif
+// #else
+//     host_addr_space_limit = address_space_limit;
+// #endif
 
     if (preload_info && *preload_info)
         for (i = 0; (*preload_info)[i].size; i++)
-- 
2.31.1

