From 862162a31da96c5890927b5f989bd245e38415eb Mon Sep 17 00:00:00 2001
From: Sun Haiyong <sunhaiyong@zdbr.net>
Date: Wed, 7 Aug 2024 03:31:29 +0000
Subject: [PATCH] cmake 3.30.2 fix for gcc 15.

---
 Utilities/cmcppdap/include/dap/network.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/Utilities/cmcppdap/include/dap/network.h b/Utilities/cmcppdap/include/dap/network.h
index 9d14f6b7..4c5868f5 100644
--- a/Utilities/cmcppdap/include/dap/network.h
+++ b/Utilities/cmcppdap/include/dap/network.h
@@ -17,6 +17,7 @@
 
 #include <functional>
 #include <memory>
+#include <cstdint>
 
 namespace dap {
 class ReaderWriter;
-- 
2.31.1

