From 40315a01a5412f3778d9aa7ea195cde8b98a9763 Mon Sep 17 00:00:00 2001
From: Sun Haiyong <sunhaiyong@zdbr.net>
Date: Tue, 24 Sep 2024 12:20:07 +0000
Subject: [PATCH] chromium 128 skia add loongarch64 define.

---
 skia/BUILD.gn | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/skia/BUILD.gn b/skia/BUILD.gn
index f1ff989..9ffccfa 100644
--- a/skia/BUILD.gn
+++ b/skia/BUILD.gn
@@ -65,6 +65,7 @@ config("skia_config") {
     "SK_DISABLE_LEGACY_TEXTURE_INFO_FUNCS",
     "SK_DISABLE_LEGACY_BACKEND_SEMAPHORE_FUNCS",
     "SK_DISABLE_LEGACY_GRAPHITE_IMAGES",
+    "SK_ENABLE_OPTIMIZE_SIZE",
   ]
 
   include_dirs = [
@@ -682,6 +683,13 @@ if (current_cpu == "x86" || current_cpu == "x64") {
   }
 }
 
+if (current_cpu == "loong64") {
+  skia_source_set("skia_opts_lasx") {
+    sources = skia_opts.lasx_sources
+    visibility = [ ":skia_opts" ]
+  }
+}
+
 skia_source_set("skia_opts") {
   cflags = []
   defines = []
@@ -720,6 +728,7 @@ skia_source_set("skia_opts") {
   } else if (current_cpu == "riscv64") {
     # Conditional and empty body needed to avoid assert() below.
   } else if (current_cpu == "loong64") {
+    deps += [ ":skia_opts_lasx" ]
     # Conditional and empty body needed to avoid assert() below.
   } else {
     assert(false, "Unknown cpu target")
-- 
2.31.1

