From 41f8c4c05ba14952f1aedbcb06bc903121514366 Mon Sep 17 00:00:00 2001
From: Sun Haiyong <sunhaiyong@zdbr.net>
Date: Mon, 8 Apr 2024 08:18:10 +0000
Subject: [PATCH] box64 for loongarch64 fix.

---
 CMakeLists.txt | 16 ++++++++--------
 1 file changed, 8 insertions(+), 8 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1a97cf0..d108cf1 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -843,14 +843,14 @@ if(LARCH64_DYNAREC)
     "${BOX64_ROOT}/src/dynarec/la64/dynarec_la64_emit_math.c"
     "${BOX64_ROOT}/src/dynarec/la64/dynarec_la64_emit_shift.c"
     "${BOX64_ROOT}/src/dynarec/la64/dynarec_la64_emit_logic.c"
-    "${BOX64_ROOT}/src/dynarec/la64/dynarec_la64_00.c"
-    "${BOX64_ROOT}/src/dynarec/la64/dynarec_la64_0f.c"
-    "${BOX64_ROOT}/src/dynarec/la64/dynarec_la64_64.c"
-    "${BOX64_ROOT}/src/dynarec/la64/dynarec_la64_66.c"
-    "${BOX64_ROOT}/src/dynarec/la64/dynarec_la64_f30f.c"
-    "${BOX64_ROOT}/src/dynarec/la64/dynarec_la64_660f.c"
-    "${BOX64_ROOT}/src/dynarec/la64/dynarec_la64_f0.c"
-    "${BOX64_ROOT}/src/dynarec/la64/dynarec_la64_f20f.c"
+    "${BOX64_ROOT}/src/dynarec/la64/dynarec_la64_00.c"
+#    "${BOX64_ROOT}/src/dynarec/la64/dynarec_la64_0f.c"
+#    "${BOX64_ROOT}/src/dynarec/la64/dynarec_la64_64.c"
+#    "${BOX64_ROOT}/src/dynarec/la64/dynarec_la64_66.c"
+#    "${BOX64_ROOT}/src/dynarec/la64/dynarec_la64_f30f.c"
+#    "${BOX64_ROOT}/src/dynarec/la64/dynarec_la64_660f.c"
+#    "${BOX64_ROOT}/src/dynarec/la64/dynarec_la64_f0.c"
+#    "${BOX64_ROOT}/src/dynarec/la64/dynarec_la64_f20f.c"
     )
 endif()
 
-- 
2.31.1

