diff -Nurp ppsspp-loongarch64.orig/Core/Config.cpp ppsspp-loongarch64/Core/Config.cpp
--- ppsspp-loongarch64.orig/Core/Config.cpp	2022-06-22 10:13:12.715242617 +0000
+++ ppsspp-loongarch64/Core/Config.cpp	2022-06-23 09:03:55.203787030 +0000
@@ -477,6 +477,8 @@ std::string CreateRandMAC() {
 static int DefaultCpuCore() {
 #if PPSSPP_ARCH(ARM) || PPSSPP_ARCH(ARM64) || PPSSPP_ARCH(X86) || PPSSPP_ARCH(AMD64)
 	return (int)CPUCore::JIT;
+#elseif PPSSPP_ARCH(LOONGARCH64)
+        return (int)CPUCore::IR_JIT;
 #else
 	return (int)CPUCore::INTERPRETER;
 #endif
