From 05c1114162dc0fd47330ecbb9e83f8e1f21fecda Mon Sep 17 00:00:00 2001
From: Sun Haiyong <sunhaiyong@zdbr.net>
Date: Mon, 30 Jun 2025 06:05:57 +0000
Subject: [PATCH 01/11] qt6 6.9.1 webengine configure add loongarch64 support.

---
 qtwebengine/configure.cmake | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/qtwebengine/configure.cmake b/qtwebengine/configure.cmake
index 86c120248e..50f8078795 100644
--- a/qtwebengine/configure.cmake
+++ b/qtwebengine/configure.cmake
@@ -306,7 +306,7 @@ qt_webengine_configure_check("supported-platform"
 )
 
 if(LINUX AND CMAKE_CROSSCOMPILING)
-    set(supported_targets "arm" "arm64" "armv7-a" "x86_64")
+    set(supported_targets "arm" "arm64" "armv7-a" "x86_64" "loongarch64")
     qt_webengine_configure_check("supported-arch"
         MODULES QtWebEngine QtPdf
         CONDITION TEST_architecture_arch IN_LIST supported_targets
-- 
2.31.1

